In [0]:
# Code to read csv file into Colaboratory:
!pip install -U -q PyDrive
from pydrive.auth import GoogleAuth
from pydrive.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials
# Authenticate and create the PyDrive client.
auth.authenticate_user()
gauth = GoogleAuth()
gauth.credentials = GoogleCredentials.get_application_default()
drive = GoogleDrive(gauth)

In [192]:
 link = 'https://drive.google.com/open?id=1Jj9sB06Lk-e6vYiCSznEWy_WdC2BmlNY' # The shareable link
fluff, id = link.split('=')
print (id) # Verify that you have everything after '='

1Jj9sB06Lk-e6vYiCSznEWy_WdC2BmlNY


In [193]:
# download stopword package from ntlk package
import nltk
nltk.download('stopwords')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [0]:
#importing librairies
%matplotlib inline
import re
import matplotlib
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score
from sklearn.multiclass import OneVsRestClassifier
from nltk.corpus import stopwords
stop_words = set(stopwords.words('english'))
from sklearn.svm import LinearSVC
from sklearn.linear_model import LogisticRegression
from sklearn.pipeline import Pipeline
import seaborn as sns
import pandas as pd
from sklearn.model_selection import train_test_split
from nltk.corpus import stopwords

In [195]:
downloaded = drive.CreateFile({'id':id}) 
downloaded.GetContentFile('train.csv')  
df = pd.read_csv('train.csv', sep='', quotechar='"', encoding='utf8')
# Dataset is now stored in a Pandas Dataframe

/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [0]:
#preprocessing
df_drop = df.drop(['faltu'], axis=1)
#df_drop['gender'].unique()

In [0]:
df1= df_drop.dropna(subset=['description', 'title'])
#df1['sub_sub_cat'].unique()

In [0]:
df2 = df1[df1['category'].isin(['automotive', 'auto exterior accessories', 'accessories',
       'luggage & bags', 'jewelry', 'watches', 'electronics',
       'laptops & netbooks', 'computers', 'cell phones',
       'socks & hosiery', 'car safety equipment',
       'flowers & food gifting', 'health & beauty', 'hair accessories',
       'luggage', 'home theatre', 'home theatre accessories',
       'computer parts & hardware', 'computer accessories',
       'electronics accessories', 'handbags', 'flower gifts', 'beauty',
       'gaming', 'cameras & video', 'jewelry & accessories',
       'kids & baby', 'health care', 'baby', 'boys clothing',
       'toys & games', 'games & puzzles', 'dolls & stuffed animals',
       'tablet & tablet accessories', 'vehicles, hobby & r/c',
       'arts & crafts supplies', 'clothing', 'home & pet',
       'school & office', 'dresses', 'home decor', 'shorts', 'activewear',
       'kitchen & dining', 'party supplies & seasonal', 'swimwear',
       'shoes', 'furniture', 'boots', 'storage & cleaning',
       'home appliances', 'sneakers & athletic shoes',
       'pretend play & dress up', 'denim', 'supplies',
       'platform & wedge shoes', 'sunglasses & frames',
       'books, movies & music', 'sports & outdoors', 'adventure sports' ]) ]


df3 = df1[df1['sub_cat'].isin(['auto exterior accessories', 'auto interior accessories',
       'watches', 'jewelry', 'scarves & shawls', 'handbags', 'sunglasses',
       'luggage & bags', 'socks & hosiery', 'hats', 'belts & buckles',
       'gloves & mittens', 'wallets & money clips', 'neckties',
       'business bags', 'umbrellas', 'eyeglasses', 'keychains',
       'hair accessories', 'necklaces', 'backpacks', 'slippers',
       'earrings', 'luggage', 'team shop', 'gaming', 'computers',
       'cell phones', 'computer parts & hardware', 'audio',
       'communications', 'cameras & video', 'headwear & kerchiefs',
       'home theatre', 'electronics accessories', 'monitors',
       'computer accessories', 'cases', 'laptops & netbooks',
       'tablet & tablet accessories', 'charms & pendants', 'flats',
       'socks', 'car safety equipment', 'handheld devices',
       'personal care', 'party supplies & seasonal',
       'underwear & intimates', 'gourmet food', 'flower gifts',
       'baby gifts', 'popcorn', 'cookie gifts',
       'coffee, tea & hot chocolate', 'chocolate gifts', 'fruit',
       'sweets', 'cakes', 'beauty', 'health care', 'sports & duffel bags',
       'sexual wellness', 'shorts', 'unisex', 'home theatre accessories',
       'trees', 'bakery gifts', 'ice cream', 'boots', 'bracelets',
       'car & marine electronics', 'desktop computers', 'networking',
       'school & office', 'power', 'software', 'home improvement',
       'tote handbags', 'meat & cheese', 'projectors & screens', 'nuts',
       'snack box', 'brownie gifts', 'sauces, spices & oils', 'soup',
       'pumps', 'dresses', 'ereaders', 'spa', 'wreaths', 'sweaters',
       'sandals', 'kids accessories', 'hair care', 'outerwear',
       'video game accessories', 'cameras', 'preserves, relish & chutney',
       'makeup & fragrance', 'pretzels', 'sneakers & athletic shoes',
       'boys clothing', 'baby', 'kids clothing', 'girls clothing',
       'home appliances', 'roses', 'home decor', 'swimwear',
       'bedding & bath', 'appetizers', 'baby clothes', 'blank media',
       'diapering', 'pet', 'rings', 'nursing & feeding', 'baby safety',
       'tops', 'baby gear', 'musical instruments', 'baby toys',
       'costumes & accessories', 'arts & crafts supplies',
       'pretend play & dress up', 'outdoor play',
       'dolls & stuffed animals', 'building blocks & sets',
       'games & puzzles', 'learning & educational toys',
       'vehicles, hobby & r/c', 'puzzles', 'board games',
       'dollhouses & accessories', 'action figures', 'racks', 'furniture',
       'boys pants & shorts', 'cheese gifts', 'pants', 'activewear',
       'denim', 'sleepwear & loungewear', 'skirts', 'suits',
       'jumpsuits & rompers', 'maternity clothing', 'uniforms',
       'kitchen & dining', 'supplies', 'food & gifts', 'patio & garden',
       'storage & cleaning', 'paper & plastic supplies',
       'decorative accents', 'active pants', 'dining',
       'retail store supplies', 'kitchen', 'holiday', 'loafers', 'rugs',
       'clothing', 'clocks', 'organization', 'nursery', 'clogs & mules',
       'platform & wedge shoes', 'oxford shoes', 'paper products',
       'adhesives & fasteners', 'ankle boots', 'golf & racquet sports',
       'cleaning', 'air care & temperature control',
       'shoe care & accessories', 'books', 'dvds & videos', 'team sports',
       'adventure sports', 'exercise & fitness', 'hiking & camping',
       'outdoor games', 'water sports', 'chairs & sofas']) ]

df4 = df1[df1['sub_sub_cat'].isin(['racks', 'bracelets', 'earrings', 'tote handbags',
       'necklaces', 'rings', 'luggage', 'socks', 'business bags',
       'gloves', 'backpacks', 'wallets', 'hobo handbags',
       'clutches & special occasion bags', 'unisex', 'shoulder handbags',
       'charms & pendants', 'crossbody bags', 'toiletry bags', 'satchels',
       'wristlets', 'sports & duffel bags', 'backpack handbags',
       'beanies', 'brooches & lapels', 'baseball caps', 'hosiery',
       'hair extensions', 'cufflinks', 'waist & fanny packs', 'mittens',
       'aviator', 'womens', 'headbands', 'watch accessories',
       'suspenders', 'ponytail holders', 'mens', 'team home & pet',
       'hair pins & clips', 'team accessories', 'nintendo', 'video games',
       'laptops & netbooks', 'software', 'cell phone accessories',
       'iphone', 'xbox', 'desktop computers', 'networking',
       'tablet & tablet accessories', 'audio accessories & components',
       'headphones', 'printers, scan & fax', 'video conferencing',
       'speakers & amps', 'cameras', 'memory', 'cases',
       'home theatre accessories', 'karaoke systems & accessories',
       'video players & recorders', 'computer accessories', 'monitors',
       'storage drives & devices', 'motherboards',
       'stereo systems, docks & radios', 'computer servers',
       'blank media', 'projectors & screens', 'graphic cards', 'power',
       'optics', 'chargers & batteries', 'bluetooth transmitters',
       'televisions', 'ipod & mp3 players', 'decorative accents',
       'handheld devices', 'wayfarer',
       'home theatre audio & video cables', 'power supplies',
       'precious stones', 'communication radios', 'telephony',
       'fans & pc cooling', 'playstation', 'ereaders',
       'video game accessories', 'digital tuners, converters & adapters',
       'headsets', 'computer cases', 'money clips', 'bath & body',
       'party supplies', 'skin care', 'round', 'bandanas',
       'floral bouquets', 'loungewear', 'popcorn gift tins',
       'cookie gift boxes', 'pc', 'plants', 'chocolate gift boxes',
       'fruit bouquets', 'beverages', 'sweets gift boxes',
       'makeup & fragrance', 'hair care', 'vitamins & supplements',
       'smoking cessation', 'suit separates', 'lubricants & moisturizers',
       'first aid', 'oral care', 'incontinence care',
       'deodorant & anti-perspirant', 'health monitors & scales',
       'medicines', 'condoms & contraceptives', 'supports & braces',
       'diet & fitness', 'gift sets', 'eye & vision care',
       'massage & relaxation', 'shaving & grooming', 'feminine care',
       'sex toys', 'sexual vitamins & supplements', 'processors', 'roses',
       'sampler set', 'gift canisters', 'cables', 'unlocked cell phones',
       'android', 'feminine health', 'tv mounts & stands',
       'homeopathic remedies', 'adapters', 'video cards',
       'car & marine gps', 'video game consoles', 'remote controls',
       'video cameras', 'enhancements & conditioning',
       'players & recorders', 'supplies', 'audio & video receivers',
       'plumbing', 'foot care', 'gourmet food gift baskets',
       'gift basket', 'baby gift baskets', 'fruit baskets',
       'cookie gift tins', 'wreaths', 'popcorn cards',
       'kitchen appliances', 'diabetes care', 'pears & fruit',
       'buttercream frosted gifts', 'cookie gift baskets',
       'popcorn gift towers', 'sweets gift baskets', 'gifts', 'orchids',
       'fruit gift boxes', 'car audio & video',
       'caramel & chocolate apples', 'mobility & accessibility',
       'sleep aids', 'cotton balls & swabs', 'family planning',
       'light therapy', 'home theatre systems', 'spa gift baskets',
       'chocolate gift tins', 'streaming media devices', 'popcorn balls',
       'ear care', 'hearing aids', 'standing spray', 'intimate care',
       'answering machines', 'boys shirts & tops', 'baby clothes',
       'nursery', 'baby gear', 'kids pants', 'boys sleep & underwear',
       'nursing & feeding', 'kids shirts & tops', 'baby safety',
       'baby health', 'diapering', 'kids dresses', 'boys pants & shorts',
       'kids sleepwear & underwear', 'girls dresses',
       'girls shirts & tops', 'girls pants, skirts & shorts',
       'boys accessories', 'baby bathing', 'kids skirts',
       'girls accessories', 'boys coats & jackets',
       'boys suits & separates', 'kids shorts', 'girls sweaters',
       'girls coats & jackets', 'kids outerwear', 'kids sweaters', 'bath',
       'kids swimwear', 'boys sweaters', 'popcorn gift baskets',
       'boys swimwear', 'girls sleep & underwear', 'girls swimwear',
       'cookie jars', 'baby gift sets', 'baby & toddler',
       'baby & toddler gifts', 'boys', 'kids suits & separates',
       'hiking & camping', 'baby walkers & entertainers',
       'costume accessories', 'costumes', 'puppets', 'dolls', 'puzzles',
       'lego', 'doll accessories', 'dollhouses & accessories',
       'kitchen & housekeeping', 'collectible games & cards', 'playmats',
       'electronic learning', 'baby mobiles', 'stuffed toys',
       'baby bouncers & rockers', 'board games', 'rattles',
       'baby jumpers & swings', 'card games', 'trains & train sets',
       'magic', 'bubbles & accessories', 'vehicles & playsets',
       'pens, pencils & markers', 'chocolate gift towers',
       'popcorn gift boxes', 'holiday', 'professionals & role playing',
       'r/c', 'organization', 'chairs & sofas', 'ballet flats',
       'food & accessories', 'teachers supplies', 'pretend electronics',
       'educational games', 'push & pull toys', 'workshops & tools',
       'dress shirts', 'leggings', 'swim bottoms', 'active tops',
       'nightgowns', 't-shirts', 'underwear', 'two-piece swimsuits',
       'bras', 'quilted & puffers', 'cover-ups', 'day dresses',
       'active shorts', 'polo shirts', 'straight leg', 'cocktail dresses',
       'vests', 'sweatshirts', 'evening dresses', 'blazers',
       'fur & faux fur', 'blouses', 'active pants', 'skinny', 'robes',
       'casual pants', 'pajamas', 'pullovers', 'one-piece swimsuits',
       'swimsuit tops', 'active jackets', 'lingerie', 'cardigans',
       'camisoles & tank tops', 'swim & board shorts',
       'wear-to-work dresses', 'turtlenecks', 'leather & faux leather',
       'raincoats', 'trench coats', 'maxi dresses', 'dress pants',
       'bootcut', 'mini skirts', 'shapewear', 'mid length skirts',
       'capes & ponchos', 'boyfriend', 'tuxedos', 'professional uniforms',
       'flare', 'cropped', 'undershirts', 'skirt suits', 'kites',
       'maternity outerwear', 'denim jackets', 'maternity shirts & tops',
       'active dresses', 'wall decor', 'dining', 'bedding', 'rugs',
       'towels', 'kitchen storage & organization', 'bedroom furniture',
       'slips', 'denim shorts', 'pillows & throws', 'kitchen',
       'food items', 'maternity dresses', 'outdoor lighting & decor',
       'desks & tables', 'lighting', 'paper products', 'cleaning',
       'adhesives & fasteners', 'blankets',
       'napkins, plates, cups & utensils', 'shipping supplies',
       'hardware & accessories', 'storage', 'dog', 'irons & steamers',
       'boards & easels', 'laundry', 'clocks', 'money & banking',
       'pant suits', 'trash supplies', 'vacuums', 'tools & equipment',
       'toilet tissue & facial tissue', 'cat', 'maxi skirts',
       'boat shoes', 'step stools & ladders', 'grilling',
       'air care & temperature control', 'workwear & safety gear',
       'calculators', 'office furniture', 'home safety & security',
       'bathroom supplies', 'tops', 'kitchen supplies', 'running shoes',
       'plus size', 'espadrilles', 'winter boots', 'ankle boots',
       'mid-calf boots', 'rain boots', 'thongs & flip flops',
       'espadrille flats', 'basketball shoes', 'over-the-knee boots',
       'hiking boots', 'platform & wedge sandals', 'golf',
       'tool storage & organization', 'gladiator sandals',
       'knee-high boots', 'kitchen & dining room furniture',
       'living room furniture', 'team clothing', 'food',
       'shredders & accessories', 'girls', 'movies', 'fiction',
       'ice & roller hockey', 'water sports', 'team shoes',
       'team kids & baby', 'bicycles & cycling gear', 'non-fiction',
       'rock climbing', 'soccer', 'baseball & softball',
       'games & outdoors activities', 'skateboard', 'football',
       'fitness accessories', 'winter sports', 'fishing', 'lacrosse',
       'fitness equipment & machines', 'basketball', 'tennis', 'running',
       'yoga & pilates' ]) ]

In [0]:
#df4.groupby('sub_sub_cat').count()

array(['auto exterior accessories', 'auto interior accessories', nan,
       'watches', 'jewelry', 'scarves & shawls', 'handbags', 'sunglasses',
       'luggage & bags', 'socks & hosiery', 'hats', 'belts & buckles',
       'gloves & mittens', 'wallets & money clips', 'neckties',
       'business bags', 'umbrellas', 'eyeglasses', 'keychains',
       'hair accessories', 'necklaces', 'backpacks', 'slippers',
       'earrings', 'luggage', 'team shop', 'gaming', 'computers',
       'cell phones', 'computer parts & hardware', 'audio',
       'communications', 'cameras & video', 'headwear & kerchiefs',
       'home theatre', 'electronics accessories', 'monitors',
       'computer accessories', 'cases', 'laptops & netbooks',
       'tablet & tablet accessories', 'charms & pendants', 'flats',
       'socks', 'car safety equipment', 'handheld devices',
       'personal care', 'party supplies & seasonal',
       'underwear & intimates', 'gourmet food', 'flower gifts',
       'baby gifts', 'popcor

In [200]:
df1.drop(df[df['gender'] == 'dress shirts'].index, inplace = True)


/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py:3940: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  errors=errors)


In [0]:

X = df_drop.description.fillna('U')
y = df_drop.gender.fillna('U')
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state = 42)

X_cat= df2.description
y_cat= df2.category
X_train1, X_test1, y_train1, y_test1 = train_test_split(X_cat, y_cat, test_size=0.3, random_state = 42)

X_sub_cat= df3.description
y_sub_cat= df3.sub_cat
X_train2, X_test2, y_train2, y_test2 = train_test_split(X_sub_cat, y_sub_cat, test_size=0.3, random_state = 42)

X_sub_sub_cat= df4.description
y_sub_sub_cat= df4.sub_sub_cat
X_train3, X_test3, y_train3, y_test3 = train_test_split(X_sub_sub_cat, y_sub_sub_cat, test_size=0.3, random_state = 42)

In [0]:
category = ['plus size', 'U', 'boys', 'womens', 'mens', 'baby & toddler', 'unisex', 'girls',
       'kids', 'baby girls', 'baby boys']

category_cat = ['automotive', 'auto exterior accessories', 'accessories',
       'luggage & bags', 'jewelry', 'watches', 'electronics',
       'laptops & netbooks', 'computers', 'cell phones',
       'socks & hosiery', 'car safety equipment',
       'flowers & food gifting', 'health & beauty', 'hair accessories',
       'luggage', 'home theatre', 'home theatre accessories',
       'computer parts & hardware', 'computer accessories',
       'electronics accessories', 'handbags', 'flower gifts', 'beauty',
       'gaming', 'cameras & video', 'jewelry & accessories',
       'kids & baby', 'health care', 'baby', 'boys clothing',
       'toys & games', 'games & puzzles', 'dolls & stuffed animals',
       'tablet & tablet accessories', 'vehicles, hobby & r/c',
       'arts & crafts supplies', 'clothing', 'home & pet',
       'school & office', 'dresses', 'home decor', 'shorts', 'shoes',
       'sunglasses & frames', 'books, movies & music']

sub_cat = ['auto exterior accessories', 'auto interior accessories',
       'scarves & shawls', 'socks & hosiery', 'hats', 'belts & buckles',
       'gloves & mittens', 'wallets & money clips', 'neckties',
       'business bags', 'umbrellas', 'eyeglasses', 'keychains',
       'hair accessories', 'necklaces', 'backpacks', 'slippers',
       'earrings', 'luggage', 'team shop', 'cell phones', 'computer parts & hardware', 'audio',
       'communications', 'cameras & video', 'headwear & kerchiefs',
       'home theatre', 'electronics accessories', 'monitors',
       'cases', 'laptops & netbooks','tablet & tablet accessories', 'charms & pendants', 'flats',
       'socks', 'car safety equipment', 'handheld devices',
       'personal care', 'party supplies & seasonal',
       'sexual wellness','home theatre accessories',
       'trees', 'bakery gifts', 'ice cream', 'boots', 'bracelets',
       'car & marine electronics', 'desktop computers', 'networking',
       'school & office', 'power', 'software', 'home improvement',
       'tote handbags', 'meat & cheese', 'projectors & screens', 'nuts',
       'snack box', 'brownie gifts', 'sauces, spices & oils', 'soup',
       'pumps', 'dresses', 'ereaders', 'spa', 'wreaths', 'sweaters',
       'sandals', 'kids accessories', 'hair care', 'outerwear',
       'video game accessories', 'cameras', 'preserves, relish & chutney',
       'makeup & fragrance', 'pretzels', 'sneakers & athletic shoes',
       'boys clothing', 'baby', 'kids clothing', 'girls clothing',
       'home appliances', 'roses', 'home decor', 'swimwear',
       'bedding & bath', 'appetizers', 'baby clothes', 'blank media',
       'diapering', 'pet', 'rings', 'nursing & feeding', 'baby safety',
       'tops', 'baby gear', 'musical instruments', 'baby toys',
       'costumes & accessories', 'arts & crafts supplies',
       'pretend play & dress up', 'outdoor play',
       'dolls & stuffed animals', 'building blocks & sets',
       'games & puzzles', 'learning & educational toys','vehicles', 'puzzles', 'board games',
       'dollhouses & accessories', 'action figures', 'racks', 'furniture',
       'boys pants & shorts', 'cheese gifts', 'pants', 'activewear',
       'decorative accents', 'active pants', 'dining',
       'retail store supplies', 'kitchen', 'holiday', 'loafers', 'rugs',
       'clothing', 'clocks', 'organization', 'nursery', 'clogs & mules',
       'platform & wedge shoes', 'oxford shoes', 'paper products',
       'adhesives & fasteners', 'ankle boots', 'golf & racquet sports',
       'cleaning', 'air care & temperature control',
       'shoe care & accessories', 'books', 'dvds & videos', 'team sports',
       'adventure sports', 'exercise & fitness', 'hiking & camping', 'water sports', 'chairs & sofas','dollhouses & accessories', 'action figures', 'racks', 'furniture',
       'boys pants & shorts'
    
]

sub_sub_cat = ['racks', 'bracelets', 'earrings', 'tote handbags',
       'necklaces', 'rings', 'luggage', 'socks', 'business bags',
       'gloves', 'backpacks', 'wallets', 'hobo handbags',
       'clutches & special occasion bags', 'shoulder handbags',
       'charms & pendants', 'crossbody bags', 'toiletry bags', 'satchels',
       'wristlets', 'sports & duffel bags', 'backpack handbags',
       'beanies', 'brooches & lapels', 'baseball caps', 'hosiery',
       'hair extensions', 'cufflinks', 'waist & fanny packs', 'mittens',
       'aviator', 'womens', 'headbands', 'watch accessories',
       'suspenders', 'ponytail holders', 'mens', 'team home & pet',
       'hair pins & clips', 'nintendo', 'video games',
       'laptops & netbooks', 'software',
       'iphone', 'xbox', 'desktop computers', 'networking',
       'tablet & tablet accessories', 'audio accessories & components',
       'headphones', 'printers, scan & fax', 'video conferencing',
       'speakers & amps', 'cameras', 'memory', 'cases',
       'home theatre accessories', 'karaoke systems & accessories',
       'video players & recorders','storage drives & devices', 'motherboards',
       'stereo systems, docks & radios', 'computer servers',
       'blank media', 'projectors & screens', 'graphic cards', 'power',
       'optics', 'chargers & batteries', 'bluetooth transmitters',
       'televisions', 'ipod & mp3 players', 'decorative accents',
       'handheld devices', 'wayfarer',
       'home theatre audio & video cables', 'power supplies',
       'precious stones', 'communication radios', 'telephony',
       'fans & pc cooling', 'playstation', 'ereaders',
       'video game accessories', 'digital tuners, converters & adapters',
       'headsets', 'computer cases', 'money clips', 'bath & body',
       'party supplies', 'skin care', 'round', 'bandanas',
       'floral bouquets', 'loungewear', 'popcorn gift tins',
       'cookie gift boxes', 'pc', 'plants', 'chocolate gift boxes',
       'fruit bouquets', 'beverages', 'sweets gift boxes',
       'makeup & fragrance', 'hair care', 'vitamins & supplements',
       'smoking cessation', 'suit separates', 'lubricants & moisturizers',
       'first aid', 'oral care', 'incontinence care',
       'deodorant & anti-perspirant', 'health monitors & scales',
       'medicines', 'condoms & contraceptives', 'supports & braces',
       'diet & fitness', 'gift sets', 'eye & vision care',
       'massage & relaxation', 'shaving & grooming', 'feminine care',
       'sex toys', 'sexual vitamins & supplements', 'processors', 'roses',
       'sampler set', 'gift canisters', 'cables', 'unlocked cell phones',
       'android', 'feminine health', 'tv mounts & stands',
       'homeopathic remedies', 'adapters', 'video cards',
       'car & marine gps', 'video game consoles', 'remote controls',
       'video cameras', 'enhancements & conditioning',
       'players & recorders', 'supplies', 'audio & video receivers',
       'plumbing', 'foot care', 'gourmet food gift baskets',
       'gift basket', 'baby gift baskets', 'fruit baskets',
       'cookie gift tins', 'wreaths', 'popcorn cards',
       'kitchen appliances', 'diabetes care', 'pears & fruit',
       'buttercream frosted gifts', 'cookie gift baskets',
       'popcorn gift towers', 'sweets gift baskets', 'gifts', 'orchids',
       'fruit gift boxes', 'car audio & video',
       'caramel & chocolate apples', 'mobility & accessibility',
       'sleep aids', 'cotton balls & swabs', 'family planning',
       'light therapy', 'home theatre systems', 'spa gift baskets',
       'chocolate gift tins', 'streaming media devices', 'popcorn balls',
       'ear care', 'hearing aids', 'standing spray', 'intimate care',
       'answering machines', 'boys shirts & tops', 'baby clothes',
       'nursery', 'baby gear', 'kids pants', 'boys sleep & underwear',
       'nursing & feeding', 'kids shirts & tops', 'baby safety',
       'baby health', 'diapering', 'boys pants & shorts',
       'kids sleepwear & underwear','girls shirts & tops', 'girls pants, skirts & shorts', 'baby bathing', 'kids skirts',
       'girls accessories', 'boys coats & jackets',
       'boys suits & separates', 'kids shorts', 'girls sweaters',
       'girls coats & jackets', 'kids outerwear', 'kids sweaters', 'bath',
       'kids swimwear', 'boys sweaters', 'popcorn gift baskets',
       'boys swimwear', 'girls sleep & underwear', 'girls swimwear',
       'cookie jars', 'baby gift sets',
       'baby & toddler gifts', 'kids suits & separates',
       'hiking & camping', 'baby walkers & entertainers',
       'costume accessories', 'costumes', 'puppets', 'dolls', 'puzzles',
       'lego', 'doll accessories', 'dollhouses & accessories',
       'kitchen & housekeeping', 'collectible games & cards', 'playmats',
       'electronic learning', 'baby mobiles', 'stuffed toys',
       'baby bouncers & rockers', 'board games', 'rattles',
       'baby jumpers & swings', 'card games', 'trains & train sets',
       'magic', 'pens, pencils & markers', 'chocolate gift towers',
       'popcorn gift boxes', 'holiday', 'professionals & role playing',
       'r/c', 'organization', 'chairs & sofas', 'ballet flats',
       'food & accessories', 'teachers supplies', 'pretend electronics',
       'educational games', 'push & pull toys', 'workshops & tools',
       'dress shirts', 'leggings', 'swim bottoms', 'active tops',
       'nightgowns', 't-shirts', 'underwear', 'two-piece swimsuits',
       'bras', 'quilted & puffers', 'cover-ups', 'day dresses',
       'active shorts', 'polo shirts', 'straight leg', 'cocktail dresses',
       'vests', 'sweatshirts', 'evening dresses', 'blazers',
       'fur & faux fur', 'blouses', 'active pants', 'skinny', 'robes',
       'casual pants', 'pajamas', 'pullovers', 'one-piece swimsuits',
       'swimsuit tops', 'active jackets', 'lingerie', 'cardigans',
       'camisoles & tank tops', 'swim & board shorts',
       'wear-to-work dresses', 'turtlenecks', 'leather & faux leather',
       'raincoats', 'trench coats', 'maxi dresses', 'dress pants',
       'bootcut', 'mini skirts', 'shapewear', 'mid length skirts',
       'capes & ponchos', 'boyfriend', 'tuxedos', 'professional uniforms',
       'flare', 'cropped', 'undershirts', 'skirt suits', 'kites',
       'maternity outerwear', 'denim jackets', 'maternity shirts & tops',
       'active dresses', 'wall decor', 'dining', 'bedding', 'rugs',
       'towels', 'kitchen storage & organization', 'bedroom furniture',
       'slips', 'denim shorts', 'pillows & throws', 'kitchen',
       'maternity dresses', 'outdoor lighting & decor',
       'desks & tables', 'lighting', 'paper products', 'cleaning',
       'adhesives & fasteners', 'blankets',
       'napkins, plates, cups & utensils', 'shipping supplies',
       'hardware & accessories', 'storage', 'dog', 'irons & steamers',
       'boards & easels', 'laundry', 'clocks', 'money & banking',
       'pant suits', 'trash supplies', 'vacuums', 'tools & equipment',
       'toilet tissue & facial tissue', 'cat', 'maxi skirts',
       'boat shoes', 'step stools & ladders', 'grilling',
       'air care & temperature control', 'workwear & safety gear',
       'calculators', 'office furniture', 'home safety & security',
       'tops', 'running shoes',
       'plus size', 'winter boots', 'ankle boots',
       'mid-calf boots', 'rain boots', 'thongs & flip flops',
       'espadrille flats', 'basketball shoes', 'over-the-knee boots',
       'ice & roller hockey', 'water sports', 'team shoes',
       'team kids & baby', 'bicycles & cycling gear', 'non-fiction',
       'rock climbing', 'soccer', 'baseball & softball',
       'skateboard', 'football','fishing', 'lacrosse', 'basketball', 'tennis', 'running',
       'yoga & pilates'
    
]

In [210]:
from sklearn.naive_bayes import MultinomialNB
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfTransformer
from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer

nb1 = Pipeline([('vect', CountVectorizer()),
               ('tfidf', TfidfTransformer()),
               ('clf', MultinomialNB()),
              ])
nb1.fit(X_train, y_train)


%time

nb2 = Pipeline([('vect', CountVectorizer()),
               ('tfidf', TfidfTransformer()),
               ('clf', MultinomialNB()),
              ])
nb2.fit(X_train1, y_train1)

%time

nb3 = Pipeline([('vect', CountVectorizer()),
               ('tfidf', TfidfTransformer()),
               ('clf', MultinomialNB()),
              ])
nb3.fit(X_train2, y_train2)

%time

nb4 = Pipeline([('vect', CountVectorizer()),
               ('tfidf', TfidfTransformer()),
               ('clf', MultinomialNB()),
              ])
nb4.fit(X_train3, y_train3)

%time

from sklearn.metrics import classification_report
y_pred = nb1.predict(X_test)

print('accuracy %s' % accuracy_score(y_pred, y_test))
print(classification_report(y_test, y_pred,target_names=category))

y_pred1 = nb2.predict(X_test1)

print('accuracy %s' % accuracy_score(y_pred1, y_test1))
print(classification_report(y_test1, y_pred1,target_names=category_cat))

y_pred2 = nb3.predict(X_test2)

print('accuracy %s' % accuracy_score(y_pred2, y_test2))
print(classification_report(y_test2, y_pred2,target_names=sub_cat))

y_pred3 = nb4.predict(X_test3)

print('accuracy %s' % accuracy_score(y_pred3, y_test3))
print(classification_report(y_test3, y_pred3,target_names=sub_sub_cat))


CPU times: user 0 ns, sys: 3 µs, total: 3 µs
Wall time: 8.11 µs
CPU times: user 3 µs, sys: 0 ns, total: 3 µs
Wall time: 7.39 µs
CPU times: user 21 µs, sys: 0 ns, total: 21 µs
Wall time: 25.3 µs
CPU times: user 2 µs, sys: 0 ns, total: 2 µs
Wall time: 6.44 µs
accuracy 0.6804818284939227


/usr/local/lib/python3.6/dist-packages/sklearn/metrics/classification.py:1437: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


                precision    recall  f1-score   support

     plus size       0.83      0.78      0.80    105274
             U       0.47      0.72      0.57     15284
          boys       0.00      0.00      0.00       946
        womens       0.00      0.00      0.00       971
          mens       0.63      0.08      0.14      6044
baby & toddler       0.73      0.24      0.37      8441
        unisex       0.00      0.00      0.00      1728
         girls       0.65      0.48      0.55     16250
          kids       0.00      0.00      0.00      1402
    baby girls       0.70      0.64      0.67     55858
     baby boys       0.50      0.82      0.62     35525

      accuracy                           0.68    247723
     macro avg       0.41      0.34      0.34    247723
  weighted avg       0.69      0.68      0.67    247723

accuracy 0.8884920181354059
                             precision    recall  f1-score   support

                 automotive       0.89      0.90      0.89 

In [0]:
prediction = pd.DataFrame(y_pred, columns=['predictions']).to_csv('/content/drive/My Drive/prediction_gender.csv')

prediction = pd.DataFrame(y_pred1, columns=['predictions_category']).to_csv('/content/drive/My Drive/prediction_category.csv')

prediction = pd.DataFrame(y_pred2, columns=['predictions_sub_cat']).to_csv('/content/drive/My Drive/prediction_sub_cat.csv')

prediction = pd.DataFrame(y_pred3, columns=['predictions_sub_sub_cat']).to_csv('/content/drive/My Drive/prediction_sub_sub_cat.csv')

68% accuracy in gender prediction
88% in category
68% in sub_cat
58% in sub_sub_cat